Sources/Virtual Currencies/VirtualCurrencyManager.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  VirtualCurrencyManager.swift
+//
+//  Created by Will Taylor on 5/28/25.
+
+import Foundation
+
+protocol VirtualCurrencyManagerType {
+    func virtualCurrencies() async throws -> VirtualCurrencies
+
+    func invalidateVirtualCurrenciesCache() async
+}
+
+actor VirtualCurrencyManager: VirtualCurrencyManagerType {
+
+    private let identityManager: IdentityManager
+    private let deviceCache: DeviceCache
+    private let backend: Backend
+    private let systemInfo: SystemInfo
+
+    init(
+        identityManager: IdentityManager,
+        deviceCache: DeviceCache,
+        backend: Backend,
+        systemInfo: SystemInfo
+    ) {
+        self.identityManager = identityManager
+        self.deviceCache = deviceCache
+        self.backend = backend
+        self.systemInfo = systemInfo
+    }
+
+    func virtualCurrencies() async throws -> VirtualCurrencies {
+        let appUserID = identityManager.currentAppUserID
+        let isAppBackgrounded = await systemInfo.isApplicationBackgrounded()
+
+        if let cachedVirtualCurrencies = await fetchCachedVirtualCurrencies(
+            appUserID: appUserID,
+            isAppBackgrounded: isAppBackgrounded
+        ) {
+            return cachedVirtualCurrencies
+        }
+
+        let virtualCurrencies = try await fetchVirtualCurrenciesFromBackend(
+            appUserID: appUserID,
+            isAppBackgrounded: isAppBackgrounded
+        )
+
+        cacheVirtualCurrencies(virtualCurrencies, appUserID: appUserID)
+
+        return virtualCurrencies
+    }
+
+    func invalidateVirtualCurrenciesCache() async {
```

Comment:
I think this method does not need to be async, and that would bubble up to the public methods simplifying usage 😄